### DISCUSSION TITLE
Laravel Octane: private properties on Controller stay in memory

### DISCUSSION THREAD

**Discussion by AlliBalliBaba** [laravel/framework#52016] [General] [answered]
I recently noticed that any private property on a Controller stays in memory across requests when using Laravel Octane. I wonder, is this intended behaviour?

```
class Controller
{
    private $count; // this property will stay in memory across requests?

    public function doSomething()
    {
        $this->count++;
        return ['count' => $this->count];
    }
}
```

**Comment by ghost** [comment]
Hi. Yes it is. Any state is kept with octane and artisan queue:work without using the --once flag.
This is why I use opcache and JIT only, no octane or daemon worker.

**Comment by AlliBalliBaba** [comment, marked as answer]
My bad this is not expected behavior. It was just a side effect of the request not terminating correctly.
Octane does indeed call `$route->flushController();` on request termintaion
